rskj-core/src/main/java/co/rsk/rpc/json/JsonResponseSizeLimiter.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc.json;
+
+import co.rsk.rpc.exception.JsonRpcResponseLimitError;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JsonResponseSizeLimiter {
+    private static final int BRACKETS_SIZE_IN_BYTES = 2;
+    private static final int COMMA_SIZE = 1;
+    private static final int JSON_FIELD_MISSING_SYMBOLS = 3;
+
+    private int acc = 0;
+    private final int max;
+    private final JsonNode root;
+    private final ObjectMapper objectMapper;
+
+    private JsonResponseSizeLimiter(int max, JsonNode root) {
+        this.max = max;
+        this.root = root;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public static int getSizeInBytesWithLimit(JsonNode node, int limit) {
+        if (node == null) {
+            return 0;
+        }
+        JsonResponseSizeLimiter limiter = new JsonResponseSizeLimiter(limit, node);
```

Comment:
ideally, the JsonResponseSizeLimiter should be as lightweight as possible. Meaning that no new objects are being created on the way. Eg. `objectMapper.writeValueAsBytes(node)` makes a new object in heap and basically duplicates that. Imagine that we have a result like this: `"result": "0x<10mb of data>"`, then `objectMapper.writeValueAsBytes()` would generate extra 10mb of data in ram.

I guess we could approximately calculate size of a node object. For instance, at runtime we could check if value is a string and simply add its length to a sum. Same for numbers and bools. In JSON, values must be one of the following data types:
```
a string
a number
an object (JSON object)
an array
a boolean
null
```
.. and for composite types like arrays/objects this can be done recursively. Obviously, this would be an approximated size, but at the very end when a response is "flushed" to OutputStream you can make a final "strict" check.

Reply:
It is only happening with fields so I though it was not a big deal and I preferred to make it simple and not to repeat the logic of check each type etc, but I will review it.